plugins/processors/awsapplicationsignals/internal/metrichandlers/aggregation_mutator.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package metrichandlers
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type aggregationType int
+
+const (
+	defaultAggregation aggregationType = iota
+	lastValueAggregation
+)
+
+// AggregationMutator is used to convert predefined ObservableUpDownCounter metrics to use LastValue metrichandlers. This
+// is necessary for cases where metrics are instrumented as cumulative, yet reported with snapshot values.
+//
+// For example, metrics like DotNetGCGen0HeapSize may report values such as 1000, 2000, 1000, with cumulative temporality
+// When exporters, such as the EMF exporter, detect these as cumulative, they convert the values to deltas,
+// resulting in outputs like -, 1000, -1000, which misrepresent the data.
+//
+// Normally, this issue could be resolved by configuring a view with LastValue metrichandlers within the SDK.
+// However, since the view feature is not fully supported in .NET, this workaround implements the required
+// conversion to LastValue metrichandlers to ensure accurate metric reporting.
+// See https://github.com/open-telemetry/opentelemetry-dotnet/issues/2618.
+type AggregationMutator struct {
+	includes map[string]aggregationType
+}
+
+func NewAggregationMutator() AggregationMutator {
+	return newAggregationMutatorWithConfig(map[string]aggregationType{
+		"DotNetGCGen0HeapSize":    lastValueAggregation,
```

Comment:
Wonder if we should just make `LastValueAggregation` as a const that can be passed in via the processor config directly from the translator and that way, we dont need any mapping here - and is also extensible for any metric keeping this logic here generic.

Reply:
The aggregation mutator (so far) is an interim solution to work around the issue that the last_value aggregation is not yet supported in `View` model in the .NET OTEL SDK. I was very reluctant to introduce this component but had no other option... We haven't observed such feature disparity in other supported language SDK.

Thus in the long term, the plan is to remove this component when .NET is ready. Making it configurable will make the deprecation harder because we will be faced with more compatibility concerns.